TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/subsystems/WebcamAprilTags.java

```diff
@@ -0,0 +1,215 @@
+import org.firstinspires.ftc.teamcode.core.lib.interfaces.Subsystem;
+import org.firstinspires.ftc.teamcode.robot.constants.AutonomousConstants;
+import org.firstinspires.ftc.vision.VisionPortal;
+import org.firstinspires.ftc.vision.apriltag.AprilTagDetection;
+import org.firstinspires.ftc.vision.apriltag.AprilTagLibrary;
+import org.firstinspires.ftc.vision.apriltag.AprilTagLibrary.Builder;
+import org.firstinspires.ftc.vision.apriltag.AprilTagProcessor;
+import org.openftc.easyopencv.OpenCvInternalCamera;
+
+import java.util.List;
+
+public class WebcamAprilTags implements Subsystem {
+    private static WebcamAprilTags instance;
+    public enum Sections{
+        BLUE_ALLIANCE_SIDE_3_NEXUSES,
+        BLUE_MIDDLE_3_NEXUSES,
+        RED_ALLIANCE_SIDE_3_NEXUSES,
+        RED_MIDDLE_3_NEXUSES,
+        BLUE_BACKWARDS_NEXUSES,
+        RED_BACKWARDS_NEXUSES,
+        BLUE_PLATAFORM_NEXUSES,
+        RED_PLATAFORM_NEXUSES,
+        UNKNOWN_SECTION
+    }
+    private Telemetry telemetry;
+    private static AprilTagProcessor aprilTag;
+    static double CAM_DIST_TO_CENTER = 0;
+    // distancia da camera até o centro do robo
+    static double CAM_ANGLE_TO_GROUND =0;
+    //angulação da câmera (0º = paralelo ao chão, 90° = olhando para cima)
+    static double actualHead;
+    static double xtrans;
+    static double ytrans;
+    Pose2d prevDetectedPose;
+    private Sections currentSection = Sections.UNKNOWN_SECTION;
+    private static VisionPortal visionPortal;
+    public static AprilTagLibrary getFeedingTheFutureTagLibrary() {
+        return (new Builder())
+                .addTag(100, "Blue Nexus Goal - Field Center - Facing Platform", 160.0D, DistanceUnit.MM)
+                .addTag(101, "Red Nexus Goal - Field Center - Facing Platform", 160.0D, DistanceUnit.MM)
+                .addTag(102, "Red Nexus Goal - Field Center - Facing Food Warehouse", 160.0D, DistanceUnit.MM)
+                .addTag(103, "Blue Nexus Goal - Field Center - Facing Food Warehouse", 160.0D, DistanceUnit.MM)
+                .addTag(104, "Blue Nexus Goal - Field Edge - Alliance Station", 160.0D, DistanceUnit.MM)
+                .addTag(105, "Blue Nexus Goal - Field Edge - Center Field", 160.0D, DistanceUnit.MM)
+                .addTag(106, "Red Nexus Goal - Field Edge - Center Field", 160.0D, DistanceUnit.MM)
+                .addTag(107, "Red Nexus Goal - Field Edge - Alliance Station", 160.0D, DistanceUnit.MM)
+                .build();
+    }
+
+    /**
+     * Add telemetry about AprilTag detections.
+     */
+    @SuppressLint("DefaultLocale")
+    public Pose2d LocateWithAprilTag() {
+
+        List<AprilTagDetection> currentDetections = aprilTag.getDetections();
+        Sections section = Sections.UNKNOWN_SECTION;
+        currentSection = Sections.UNKNOWN_SECTION;
+
+        // Step through the list of detections and display info for each one.
+        if (currentDetections.size() > 0) {
+            double distX = 0;
```

Comment:
as a suggestion, create an object that will return these three values and then incapsulate this in a method to get it's value

Reply:
Did it by creating a class called AprilTagFieldData to hold X, Y and Section values for each tag and adding methods to WebcamAprilTags to acess said values